runtime/mux.go

```diff
@@ -244,8 +284,12 @@ func (s *ServeMux) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 			components[l-1], verb = lastComponent[:idx], lastComponent[idx+1:]
 		}
 
-		pathParams, err := h.pat.Match(components, verb)
+		pathParams, err := h.pat.MatchAndEscape(components, verb, s.unescapingMode)
 		if err != nil {
+			if err == ErrMalformedSequence {
```

Comment:
What do you think about using `errors.Is` instead?
```suggestion
			if errors.Is(err,  ErrMalformedSequence) {
```

Reply:
100% for it, was just matching the rest of the source. Will update it and propagate a descriptive error message.

Reply:
I changed to errors.As and propagating the real error (which indicates the bad escape sequence), let me know if that looks okay.